File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/persistence/candidate/JpaCandidateRepositoryImpl.java
package kyonggiyo.persistence.candidate;

import kyonggiyo.common.exception.GlobalErrorCode;
import kyonggiyo.common.exception.NotFoundException;
import kyonggiyo.domain.candidate.Candidate;
import kyonggiyo.domain.candidate.Status;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class JpaCandidateRepositoryImpl implements CandidateRepository {

    private final CandidateJpaRepository jpaRepository;

    @Override
    public Candidate getById(Long id) {
        return jpaRepository.findById(id)
                .orElseThrow(() -> new NotFoundException(GlobalErrorCode.NOT_FOUND_ENTITY_EXCEPTION));
    }

    @Override
    public Slice<Candidate> findAllByStatus(Status status, Pageable pageable) {
        return jpaRepository.findAllByStatus(status, pageable);
    }

    @Override
    public Candidate save(Candidate candidate) {
        return jpaRepository.save(candidate);
    }

    @Override
    public void deleteById(Long id) {
        jpaRepository.deleteById(id);
    }

}

File: kyonggiyo-infrastructure/src/main/java/kyonggiyo/candidate/repository/JpaCandidateRepositoryImpl.kt
package kyonggiyo.candidate.repository

import kyonggiyo.application.candidate.domain.entity.Candidate
import kyonggiyo.application.candidate.domain.vo.Status
import kyonggiyo.common.exception.GlobalErrorCode
import kyonggiyo.common.exception.NotFoundException
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Slice
import org.springframework.stereotype.Repository

@Repository
class JpaCandidateRepositoryImpl(
        private val jpaRepository: CandidateJpaRepository
) : CandidateRepository {

    override fun getById(id: Long): Candidate {
        return jpaRepository.findById(id)
                .orElseThrow { NotFoundException(GlobalErrorCode.NOT_FOUND_ENTITY_EXCEPTION) }
    }

    override fun findAllByStatus(status: Status, pageable: Pageable): Slice<Candidate> {
        return jpaRepository.findAllByStatus(status, pageable)
    }

    override fun save(candidate: Candidate): Candidate {
        return jpaRepository.save(candidate)
    }

    override fun deleteById(id: Long) {
        jpaRepository.deleteById(id)
    }

}
